conda_smithy/feedstocks.py

```diff
@@ -87,12 +86,14 @@ def clone_feedstock(feedstock_gh_repo, feedstocks_dir):
     clone_directory = os.path.join(feedstocks_dir, repo.name)
     if not os.path.exists(clone_directory):
         print(f"Cloning {repo.name}")
-        clone = Repo.clone_from(repo.clone_url, clone_directory)
-        clone.delete_remote("origin")
-    clone = Repo(clone_directory)
-    if "upstream" in [remote.name for remote in clone.remotes]:
-        clone.delete_remote("upstream")
-    clone.create_remote("upstream", url=repo.clone_url)
+        clone = pygit2.clone_repository(repo.clone_url, clone_directory)
+        clone.remotes.delete("origin")
```

Comment:
So wait, now we can clone? I thought we were using `git` subprocesses for this.

Reply:
Yes, libgit2 can clone directly. However, as noted in the PR discussion this (as implemented here) works for public HTTPS repos only. Using libgit2 to clone via SSH would require fixed libgit2 on conda *and* explicit credential handling which would be a lot of code and not 100% equivalent to external git anyway.